app/src/main/java/net/squanchy/signin/SignInActivity.java

```diff
@@ -48,6 +67,30 @@ private GoogleApiClient connectToGoogleApis() {
                 .build();
     }
 
+    private void setBottomSheetCallbackOn(View bottomSheet) {
+        BottomSheetBehavior.from(bottomSheet)
+                .setBottomSheetCallback(new BottomSheetBehavior.BottomSheetCallback() {
+                    @Override
+                    public void onStateChanged(@NonNull View bottomSheet, int newState) {
+                        switch (newState) {
+                            case BottomSheetBehavior.STATE_HIDDEN:
+                                finish();
+                                break;
+                        }
+                    }
+
+                    @Override
+                    public void onSlide(@NonNull View bottomSheet, float slideOffset) {
+                        // no op
+                    }
+                });
+    }
+
+    private void setupWindowParameters() {
```

Comment:
Yeah I debated what I should call it but this is the best I could come up with. Promise I'll improve the name if I come up with something better ;)